### DISCUSSION TITLE
How to fix CORB when i use youtube embedded video url?

### DISCUSSION THREAD

**Discussion by ivanyufen** [igordanchenko/yet-another-react-lightbox#16] [Q&A] [answered]
hi, im using this library and got a warning like this

Cross-Origin Read Blocking (CORB) blocked cross-origin response.....

i use youtube embed video url like: https://www.youtube.com/embed/{videoId}

is there any way to solve this? thank you :)

**Comment by igordanchenko** [comment, marked as answer]
The Video plugin supports raw video files (.mp4, .webm, etc.), but it sounds like you are trying to use YouTube iframe embed. You can render iframe video as a custom slide:

https://codesandbox.io/s/yet-another-react-lightbox-16-bwx8nc

```jsx
<Lightbox
  open={open}
  close={() => setOpen(false)}
  slides={[
    {
      type: "youtube",
      src: "https://www.youtube.com/embed/aqz-KE-bpKQ",
      title: "Big Buck Bunny",
      width: 2160,
      height: 1215
    }
  ]}
  render={{
    slide: ({ type, src, width, height, title }, offset, rect) =>
      type === "youtube" ? (
        <iframe
          width={Math.min(
            width,
            rect.width,
            (width * rect.height) / height
          )}
          height={Math.min(
            height,
            rect.height,
            (height * rect.width) / width
          )}
          src={src}
          title={title}
          frameBorder="0"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
        ></iframe>
      ) : null
  }}
/>
```

**Reply by ivanyufen** [reply]
hi, is there any way to use the default style on the else logic? because from the code, it is just null returned when it's not type === "youtube"

my content consists of aws images and youtube embed video, and i want to show the images one without custom slide

**Reply by igordanchenko** [reply]
You don't have to do anything special here as it already works the way you want it to. The lightbox executes its default rendering algorithm when render function returns null. I've added an image slide to this example to demonstrate this:

https://codesandbox.io/s/yet-another-react-lightbox-16-bwx8nc

**Reply by ivanyufen** [reply]
i see, that's truly a big help. thanks! :)

**Reply by ivanyufen** [reply]
hi, i just build my project and got an error React.createContext is not a function on this package. Perhaps it still using the old react version?

**Reply by igordanchenko** [reply]
What version of React are you using in your project?

**Reply by ivanyufen** [reply]
I'm using react 17.0.2 and next 11.1.4

**Reply by igordanchenko** [reply]
This library uses React as peer dependency, so it'll use whatever React version you have in your project. There should be no issues with React 17. However, you description does not provide sufficient details to troubleshoot. I can take a look if you can provide an isolated sample project showing this error. An example on CodeSandbox would be great.

**Reply by igordanchenko** [reply]
@ivanyufen, regarding the "React.createContext is not a function" error - please make sure you are using an up-to-date node version. See similar issue [here](https://github.com/igordanchenko/yet-another-react-lightbox/issues/20#issuecomment-1242109047).
